### PR TITLE
Allow fast boot disable on region

### PR DESCRIPTION
**Pull request by lbajolet-hashicorp**
The fast-launch configuration allows setupping fast-launch for Windows virtual machines, either with a common configuration, or with a region-by-region configuration.
However if someone wants to only enable fast-launch on a subset of regions, there wasn't any way to express that, leading to either a waste of resources, or a need to manually cleanup the extra snapshots created.

Therefore this commit adds an extra trilean to the regional configuration for fast-launch, so it can be explicitly disabled on selected regions.

Closes: #506

### CODE REVIEW

File: builder/ebs/fast_launch_setup.go [hunk 1]
```diff
@@ -17,6 +19,16 @@ import (
...
 type FastLaunchTemplateConfig struct {
+	// EnableFalseLaunch allows you to specify if fast launch should be
+	// enabled on a region-by-region basis
+	//
+	// If unset, this will assume the default behaviour, i.e. the other
+	// options will imply that this is true.
+	//
+	// If this is explicitly set to `false` however, fast-launch will be
+	// disabled for the specified region, and options other than the region
+	// will be ignored.
```
Comment (nywilken) [inline builder/ebs/fast_launch_setup.go:30]: Small rewrite to fix typo and help clarify the intent. 
Is the default behavior always true or based on some condition? 

```suggestion
	// Enable fast launch allows you to disable fast launch settings on the region level.
	//
	// If unset, the default region behavior will be assumed.
	// Using other fast launch options, while unset, will imply enable_fast_launch to be true.
	//
	// If this is explicitly set to `false` fast-launch will be
	// disabled for the specified region and all other options besides region
	// will be ignored.
```

File: builder/ebs/step_prepare_fast_launch_template.go [hunk 2]
```diff
@@ -49,6 +54,14 @@ func (s *stepPrepareFastLaunchTemplate) Run(ctx context.Context, state multistep
 	for _, templateSpec := range s.RegionTemplates {
 		region := templateSpec.Region
 
+		if templateSpec.EnableFalseLaunch == config.TriFalse {
+			log.Printf("[INFO] fast-launch explicitly disabled for region %q", region)
+			templateIDsByRegion[region] = TemplateSpec{
+				Enabled: false,
```
Comment (nywilken) [inline builder/ebs/step_prepare_fast_launch_template.go:60]: I'm not to clear on how this works by default but from what I gather (looking at the issue) fast launch seems to be enabled by default on the region. Is that correct? 

If so, does the user need any particular permissions to disable on the region?

**Review by nywilken — COMMENTED** [review]
Overall this looks good. I left a suggestion on the documentation and a question about permissions, if needed.

File: builder/ebs/fast_launch_setup.go (hunk 1 above)
Comment (lbajolet-hashicorp) [reply builder/ebs/fast_launch_setup.go:30]: Yeah so the default behaviour is implicitly true if there's a `region_launch_templates` defined with at least one of the mandatory attributes besides `region` (refer to the [`Prepare`](https://github.com/hashicorp/packer-plugin-amazon/blob/main/builder/ebs/fast_launch_setup.go#L43) implementation for the structure for more info).
If explicitely set to false, this just disables fast-launch for the region, as is requested in the linked issue.

File: builder/ebs/step_prepare_fast_launch_template.go (hunk 2 above)
Comment (lbajolet-hashicorp) [reply builder/ebs/step_prepare_fast_launch_template.go:60]: No permissions necessary, we enable fast-launch for each region in a later step, so if you don't want it on a particular region, we don't request it be enabled

**Review by nywilken — APPROVED** [review]
Thanks for the clarification. LGTM

**Pull request merged**